docs/adr/0003-outsource-user-management.md

```diff
@@ -0,0 +1,89 @@
+# 3. Outsource User Management
+
+* Status: accepted <!-- optional -->
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1 <!-- optional -->
+* Date: 2020-12-09 <!-- optional -->
+
+Technical Story: [Skip account-service by talking to CS3 user-api](https://github.com/owncloud/ocis/pull/1020) <!-- optional -->
+
+## Context and Problem Statement
+
+To attach metadata like shares to users ownCloud relies on persistent, non-reassignable, unique identifiers for users (and files). Email and username can change when a user changes his name. But even the OIDC sub+iss combination may change when the IdP changes. While there is [an account porting protocol](https://openid.net/specs/openid-connect-account-porting-1_0.html) that describes how a relying party such as ownCloud should should behave, it still requires the RP to maintain its own user identifiers.
+
+## Decision Drivers <!-- optional -->
+
+* OCIS should be a single binary that can run out of the box without external dependencies like an LDAP server.
+* Time: we want to build a release candidate asap.
+
+## Considered Options
+
+* Accounts service wraps LDAP
+* GLauth wraps accounts service
+
+## Decision Outcome
+
+Chosen option: "Move accounts functionality to GLauth and name it accounts", by moving the existing accounts service file based persistence to glauth and use it as a drop in replacement for an LDAP server. The reverse index and web ui existing in the accounts service will move as well in order to make glauth a standalone, small scale user management with write capabilities.
+
+### Product summary
+- GLauth is a drop in user management for small scale deployments.
+- OCIS admins can either use the web ui to manage users in glauth or use existing tools in their IDM.
+- We hide the complexity by embedding OpenID Provider, an LDAP server and a user management web ui.
+
+### Resulting deployment options
+- Single binary: admin can manage users, groups and roles using the built in web ui (glauth)
+- External LDAP: OCIS admin needs to use existing tool to manage users
+- Separate OCIS and LDAP admin: OCIS admin relies on the LDAP admin to manage users
+
+### Resulting technical implications
+- add graphapi to glauth so the ocis web ui can use it to manage users
```

Comment:
I thought the web ui should move to GLAuth? Embedding it in oC Web could be done by wrapping it into an oC Web extension (need to check if that's possible. I don't want an iframe solution).

Reply:
We may not be able to theme the glauth web ui as we like it. Also the tight integration we have in the current implementation is awesome / slick. We need to embed glauth anyway. The first step would be to treat the current accounts service as _the drop in ldap server_ with a web ui to manage users. It would mean merging ocis-accounts and ocis-glauth.
The next step would be to add the graph api to glauth. then we can build a web ui that uses glauth to manage accounts.

This accounts-management service would be a web ui to manage accounts via the graph api. It could be part of glauth or we use an embedded, tightly integrated version for ocis-web that follows our theme.

The final step would be a graphapi implementation that is backed by a real ldap server. I'd put that into glauth because it already has several backends. Having a graphapi <-> ldap proxy would fit GLauth I think.

But in order to have a truly embeddable web ui I think glauth shoul start with the graph api. It can then have a web ui or projects that want to embed it can build the ui in the way they want. The kopano guys @fbartels @longsleep might want to use react to embed it into konnectd.

Reply:
added as a bullet point, because I agree: keep the ui in GLAuth